builder/common/run_config.go

```diff
@@ -606,6 +614,13 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		errs = append(errs, fmt.Errorf("either instance_type or "+
 			"spot_instance_types must be specified, not both"))
 	}
+	
+	if c.FleetTags != nil {
+		if c.SpotPrice == "" || c.SpotPrice == "0" {
+			errs = append(errs, fmt.Errorf(
+				"fleet_tags should not be set when not requesting a spot instance"))
```

Comment:
Why is this? The SpotPrice is used inside of the fleet request

Reply:
This check is there as FleetTags wouldn't be used if we're not requesting a spot fleet. To check whether we are requesting a spot fleet or not, we check the SpotPrice value, which is required in that case. This is actually copied from an already existing check for SpotTags, a few lines down:

```
if c.SpotTags != nil { 
    if c.SpotPrice == "" || c.SpotPrice == "0" {
        errs = append(errs, fmt.Errorf(
            "spot_tags should not be set when not requesting a spot instance"))
     }
}
```

Reply:
Oh! I was reading the conditional backwards. I thought you were having validation fail if the user _is_ using a spot price. Yes, I think this is fine.